File: Leetcode/Top Interview Questions/14.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        firstWord = strs[0]
        i = 0
        while i < len(firstWord):
            for str in strs:
                if str[i] != firstWord[i]:
                    return "" if i == 0 else firstWord[:i]
            i += 1
        return firstWord

File: Leetcode/Top Interview Questions/14.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
  strs = strs.sort((a, b) => a.length - b.length);

  for (let i = strs[0].length; i > 0; i--) {
    const check = strs[0].slice(0, i);

    let j = 1;
    while (j < strs.length) {
      if (strs[j].slice(0, i) !== check) {
        break;
      }
      j++;
    }

    if (j == strs.length) {
      return check;
    }
  }
  return "";
};
